include/mapnik/text/harfbuzz_shaper.hpp

```diff
@@ -134,14 +141,16 @@ static void shape_text(text_line & line,
             double max_glyph_height = 0;
             for (unsigned i=0; i<num_glyphs; ++i)
             {
-                auto const& gpos = positions[i];
-                auto const& glyph = glyphs[i];
+                glyph_face_info const* gfi = gfaceinfos.find(i) != gfaceinfos.end()? &gfaceinfos[i]: nullptr;
```

Comment:
Can you change this logic so that it doesn't use a raw pointer? Could just making an if statement here be more clear?